### PR TITLE
*: support analyze all indices statement

### PR DESCRIPTION
**Pull request by fipped**
Add a statement `analyze table tbl_name index` to support analyze all indices for a table.
To #5373 
PTAL @lamxTyler @hanfei1991 @coocood

### CODE REVIEW

File: plan/planbuilder.go [hunk 1]
```diff
@@ -488,11 +488,24 @@ func (b *planBuilder) buildAnalyzeIndex(as *ast.AnalyzeTableStmt) Plan {
 	return p
 }
 
+func (b *planBuilder) buildAnalyzeAllIndex(as *ast.AnalyzeTableStmt) Plan {
+	p := &Analyze{}
+	tblInfo := as.TableNames[0].TableInfo
+	for _, idx := range tblInfo.Indices {
```
Comment (shenli) [inline plan/planbuilder.go]: If we use primary key as handle, it will not appear in tblInfo.Indices.

File: ast/stats.go [hunk 2]
```diff
@@ -26,6 +26,7 @@ type AnalyzeTableStmt struct {
 
 	TableNames []*TableName
 	IndexNames []model.CIStr
+	IndexFlag  bool
```
Comment (shenli) [inline ast/stats.go]: Add comment for this.

**Comment by shenli** [comment]
@fipped Please sign the CLA.

File: plan/planbuilder.go (hunk 1 above)
Comment (fipped) [reply plan/planbuilder.go]: Yes, it is expected.

File: plan/cbo_test.go [hunk 3]
```diff
@@ -300,6 +300,12 @@ func (s *testAnalyzeSuite) TestAnalyze(c *C) {
 
 	testKit.MustExec("create table t3 (a int, b int)")
 	testKit.MustExec("create index a on t3 (a)")
+
+	testKit.MustExec("create table t4 (a int, b int, c int)")
+	testKit.MustExec("create index a on t4 (a)")
+	testKit.MustExec("create index b on t4 (b)")
+	testKit.MustExec("insert into t4 (a,b,c) values (1,1,1),(1,2,1),(1,3,1),(1,4,1),(2,5,2),(2,6,2),(2,7,2),(2,8,2)")
```
Comment (winoros) [inline plan/cbo_test.go]: Don't need to insert data and don't need a new table, just add one more index to t3 and test on it.

**Comment by shenli** [comment]
LGTM

File: plan/planbuilder.go (hunk 1 above)
Comment (lamxTyler) [inline plan/planbuilder.go]: We should check if the state of index is public.

**Pull request merged**